frontend/src/components/CurrentBooking.tsx

```diff
@@ -0,0 +1,182 @@
+import './CurrentBooking.css';
+import {
+    Box,
+    Card,
+    CardActions,
+    CardContent,
+    Collapse,
+    Grid,
+    IconButton,
+    List,
+    Typography
+} from '@mui/material';
+import { Booking } from '../types';
+import React, { useState, useEffect } from 'react';
+import { getBookings } from '../services/bookingService';
+import { ExpandLess, ExpandMore, Group } from '@mui/icons-material';
+
+function getBookingRoomName(booking: Booking) {
+    return booking.room.name;
+}
+
+function getBookingTimeLeft(booking: Booking) {
+    let endTime = new Date(Date.parse(booking.endTime));
+    return getTimeDifference(endTime, new Date());
+}
+
+function getTimeDifference(startTime: Date, endTime: Date) {
+    return Math.floor((startTime.getTime() - endTime.getTime()) / 1000 / 60);
+}
+
+function areBookingsFetched(bookings: Booking[]) {
+    return Array.isArray(bookings) && bookings.length > 0;
+}
+
+function getCapacity(booking: Booking) {
+    return booking.room.capacity;
+}
+
+function getFeatures(booking: Booking) {
+    let featureArray = booking.room.features;
+    let featuresDisplay = [];
+
+    // Format booking.room features
+    if (featureArray) {
+        for (let feature = 0; feature < featureArray.length; feature++) {
+            featuresDisplay.push(featureArray[feature]);
+            if (feature !== featureArray.length - 1) {
+                featuresDisplay.push(', ');
+            }
+        }
+    }
+    return featuresDisplay;
+}
+
+function CurrentBooking() {
+    const [bookings, setBookings] = useState<Booking[]>([]);
+
+    useEffect(() => {
+        getBookings().then(setBookings);
+    }, []);
+
+    const [expandedFeatures, setExpandedFeatures] = React.useState('false');
+
+    const handleFeaturesCollapse = (
+        event: React.MouseEvent<HTMLElement>,
+        booking: Booking
+    ) => {
+        setExpandedFeatures(
+            expandedFeatures === booking.id ? 'false' : booking.id
+        );
+    };
+
+    return !areBookingsFetched(bookings) ? null : (
```

Comment:
Osaakohan @vlunn tai @Mikkooo sanoo onko tää nyt oikee tapa renderöitä react komponentti? Eli jos käyttäjälle ei löydy tämän hetkistä varausta palauttaa CurrentBooking komponentti nullin

Reply:
Joo, nullin palauttaminen on ok ja käytetty tapa piilottaa komponentti.

Ehkä tässä esimerkissä kun koko komponentti on tuon ehdon takana voisi laittaa ennen tuota returnia toisen returnin :

if (!areBookingsFetched(bookings)) return null;
return ( 
  // ...itse komponentti
  
  https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering